File: Assets/GoldSystem/GainXP.js
#pragma strict

private var levelUpScript : LevelUpSystem;

function Start () 
{	
	levelUpScript = GameObject.Find("Main Camera").GetComponent(LevelUpSystem);


}

function OnCollisionEnter2D()
{
    		print("asdf");
        //Destroy(col.gameObject);
}

function OnDestroy () {
        levelUpScript.currentXP += 10;

        print("Script was destroyed");

    }

File: Assets/Standard Assets/2D/Scripts/LevelUpSystem/GainXP.cs

using UnityEngine;
using System.Collections;

public class GainXP : MonoBehaviour {
	
	
	private LevelUpSystem levelUpScript;
	private killCounter kills;

	
	void  Start (){	
		levelUpScript = GameObject.Find("Main Camera").GetComponent<LevelUpSystem>();
		kills = GameObject.Find("Main Camera").GetComponent<killCounter>();
	}
	
	void  OnCollisionEnter2D (){
		//print("asdf");
		//Destroy(col.gameObject);
	}
	
	void  OnDestroy (){
		levelUpScript.currentXP += 10;

		kills.incScore ();
		//print("Script was destroyed");
		
	}
}